Design-Docs/standardizing-cluster-infrastructure.md

```diff
@@ -0,0 +1,277 @@
+
+To achieve this, the Cluster API provider integrations have to be standardized in some way. Cluster operators should only configure objects that are independent of the provider. For example, instead of having to configure the `ProviderMachine` object of two different providers, the cluster operator should only configure an SCS machine object, which is then translated into the object of the relevant provider. This is not meant to change the provider integrations themselves, but rather to have an additional translation layer.
+
+### Proposal 3: Users should be able to define applications independently of the cluster infrastructure
+
+SCS should have a repository with application configurations. The configurations in this repository should not depend on the provider, as this would create a lot of additional maintenance work. Additionally, it would not be possible to migrate applications easily from one provider to another.
+
+To achieve independence of application configurations from the cluster infrastructure, there are mainly two tasks that have to be done: standardizing the CCM, which currently requires provider-specific configurations of service objects, and standardizing the naming of storage classes.  
+
+## User stories
+
+The persona we call “user” (using SCS clusters) is only supposed to work with applications and should not worry at all about the cluster infrastructure below. Cluster operators define and manage clusters based on pre-defined cluster stacks that are implemented by infrastructure providers.
+
+### User story 1: DevOps expert configuring applications on SCS infrastructure
+
+A DevOps expert wants to add a Traefik ingress to the SCS application stack. As the ingress needs a load balancer, he defines a service of type load balancer. In the SCS documentation, there is a section describing how to implement a load balancer and what kind of parameters can be set, such that the DevOps expert does not need a long time to finish the configuration. This configuration is now valid for all cloud providers and can be used for every cluster on different European cloud providers. The maintainability of SCS applications is ensured by this independence of the application and cluster layer.
+
+### User story 2: Cloud provider implementing SCS stack
+
+After the cloud provider implemented the SCS cluster stack, all applications are ready to be installed and work perfectly. Therefore, the cloud provider does not have to invest time in configuring applications but can rely on the pre-defined and well-maintained SCS stack. Its users can profit from the additional value generated by this feature of ready-to-install applications. E2E tests ensure that everything has been implemented correctly.
+
+### User story 3: User wanting to implement a multi-cloud
```

Comment:
What is "a multi-cloud"?

Reply:
Good point. I think it would be good if we could define the SCS globally so that we have the same understanding of it everywhere.

Reply:
We mean a multi-provider cloud by that.